### PR TITLE
Add Minimal and MicroOS images to openSUSE:Factory:Staging (try #2)

### PR DESCRIPTION
**Pull request by Vogtinator**
Compared to #157 I had to fix a bug in the generator (`flavor_filter` broken) and the rsync location should be fixed by using flavor-specific `folder` attributes instead of `iso_path="*product*"`. Here running the rsync commands works fine and the openQA settings look ok as well.

### CODE REVIEW

File: script/cfg.py [hunk 1]
```diff
@@ -35,8 +35,12 @@
 read_files_repo_link3 = '''cp __envdir/REPOLINK/files_iso*.lst __envsub/
 '''
 
-def rsync_fix_dest(distri, version, staging):
+def rsync_fix_dest(distri, version, staging, use_staging_patterns ):
```
Comment (Martchus) [inline script/cfg.py:38]:
```suggestion
def rsync_fix_dest(distri, version, staging, use_staging_patterns):
```

File: script/cfg.py [hunk 2]
```diff
@@ -52,19 +56,19 @@ def rsync_commands(checksum):
         echo "rsync --timeout=3600 -tlp4 --specials PRODUCTISOPATH/${iso_folder[$flavor]}*$src.sha256 /var/lib/openqa/factory/other/$dest.sha256"'''
     return res
 
-rsync_iso = lambda distri, version, archs, staging, checksum, repo0folder: '''
+rsync_iso = lambda distri, version, archs, staging, checksum, repo0folder, use_staging_patterns : '''
```
Comment (Martchus) [inline script/cfg.py:59]:
```suggestion
rsync_iso = lambda distri, version, archs, staging, checksum, repo0folder, use_staging_patterns: '''
```

File: script/scriptgen.py [hunk 3]
```diff
@@ -926,6 +959,25 @@ def gen_print_openqa(self, f):
...
+                self.p(
+                    ' [ -z "$iso1" ] || echo " ISO_1=${iso1//${flavor_iso1[${flavor#Staging-}]}/Staging:__STAGING-${flavor_iso1[${flavor#Staging-}]}} \\\\"',
+                    f,
+                )
+            else:
+                self.p(' [ -z "$iso1" ] || echo " ISO_1=$iso1 \\\\"', f)
+        if len(self.staging_pattern) > 0:
+            self.p(' staging_pattern="${flavor_staging[${flavor#Staging-}]}"', f)
+            self.p(
+                " [ -z $staging_pattern ] || destiso=${destiso//$staging_pattern/Staging:__STAGING-$staging_pattern}", f
```
Comment (Martchus) [inline script/scriptgen.py:978]:
```suggestion
                ' [ -z "$staging_pattern" ] || destiso=${destiso//$staging_pattern/Staging:__STAGING-$staging_pattern}', f
```

**Review by Martchus — APPROVED** [review]

**Review by andrii-suse — APPROVED** [review]
Looks good, the only remaining bit is to make style check happy by moving `f,` to own line here:

```
black --check --fast --diff script/scriptgen.py 
--- script/scriptgen.py	2022-11-07 10:50:45.844758 +0000
+++ script/scriptgen.py	2022-11-07 10:52:09.973597 +0000
@@ -973,11 +973,12 @@
             else:
                 self.p(' [ -z "$iso1" ] || echo " ISO_1=$iso1 \\\\"', f)
         if len(self.staging_pattern) > 0:
             self.p(' staging_pattern="${flavor_staging[${flavor#Staging-}]}"', f)
             self.p(
-                ' [ -z "$staging_pattern" ] || destiso=${destiso//$staging_pattern/Staging:__STAGING-$staging_pattern}', f
+                ' [ -z "$staging_pattern" ] || destiso=${destiso//$staging_pattern/Staging:__STAGING-$staging_pattern}',
+                f,
             )
 
         if self.repolink and not self.iso_5:
             self.p(cfg.openqa_call_legacy_builds_link, f)
         if self.legacy_builds or (self.repolink and not self.iso_5):
would reformat script/scriptgen.py

```

**Comment by Vogtinator** [comment]
Should be good to go, let's give it a try next week?

**Comment by Vogtinator** [comment]
Works! I resynced D and hooked up some jobs to the product. Now the real work starts...

**Comment by mloviska** [comment]
Is this error related to the changes?

```
Generating scripts for t/abs/OBS:Server:Unstable
OK
Generating scripts for t/abs/Test1
OK
Generating scripts for t/abs/TestExtractAsRepo1
OK
Generating scripts for t/abs/TestIsoHdd-1
OK
Generating scripts for t/abs/TestOnlyIsosPost
OK
Generating scripts for t/abs/TestVersionFromMedia1
OK
Generating scripts for t/ibs/SUSE:SLE-12-SP5:Update:Products:SLERT
Traceback (most recent call last):
  File "/home/martin/git/openqa-trigger-from-obs/script/scriptgen.py", line 1250, in <module>
    ret = gen_files(args.project)
  File "/home/martin/git/openqa-trigger-from-obs/script/scriptgen.py", line 1226, in gen_files
    batch.gen_if_not_customized(path, "print_rsync_iso.sh")
  File "/home/martin/git/openqa-trigger-from-obs/script/scriptgen.py", line 466, in gen_if_not_customized
    self.gen_print_rsync_iso(f)
  File "/home/martin/git/openqa-trigger-from-obs/script/scriptgen.py", line 793, in gen_print_rsync_iso
    cfg.rsync_iso(
  File "/home/martin/git/openqa-trigger-from-obs/script/cfg.py", line 71, in <lambda>
    ''' + rsync_fix_dest(distri, version, staging, use_staging_patterns) + '''
TypeError: rsync_fix_dest() takes 3 positional arguments but 4 were given
make: *** [Makefile:31: test_regen_all] Error 1
```

**Comment by Vogtinator** [comment]
> TypeError: rsync_fix_dest() takes 3 positional arguments but 4 were given

That looks strange, the definition here accepts 4 arguments

**Pull request merged**